Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -336,6 +359,7 @@ void FunctionDefinitionWriter::visit(AST::StructType& structType)
 
 void FunctionDefinitionWriter::visit(AST::ReferenceType& type)
 {
+    m_stringBuilder.append("device ");
```

Comment:
I think this is wrong.

(I'm assuming "Reference type"s correspond to the concept of the same name in the spec: https://gpuweb.github.io/gpuweb/wgsl/#ref-ptr-types

If so, then a reference can refer to an lvalue in any address space, not just the `device` address space.

The reference type itself probably needs to remember what address space it's point into to emit the correct MSL.

Reply:
That's fair enough, I'll add a FIXME.